associations/serializers/association.py

```diff
@@ -37,6 +36,41 @@ class Meta:
         model = Association
         fields = ('id', 'name', 'logo', 'pages', 'marketplace', 'library', 'groups')
 
-    def create(self, validated_data):
-        instance = Association.objects.create(**validated_data)
+    def update(self, instance, validated_data):
+        groups_data = validated_data.pop('groups')
+        instance = super().update(instance, validated_data)
+
+        ids = set()
+
+        for group_data in groups_data:
+            id = dict(group_data)["id"]
+
+            try:
+                group = Group.objects.get(pk=id)
+            except Exception: # help requested : too broad catch
+                id = -1
+
+            if id == -1:
+                group_data.pop("id", None)
+
+                group = GroupSerializer().create(group_data) # help requested : no data validation
+                group.save()
+                instance.groups.add(group)
+                ids.add(group.id)
+
+            else:
+                serialized = GroupSerializer(group, data=group_data)
+
+                if serialized.is_valid():
+                    serialized.save()
+                    ids.add(group.id)
+
+            if not serialized.is_valid():
+                raise Exception(serialized.errors)
+
+        for group in instance.groups.all():
+            if group.id not in ids:
+                group.delete()
+
+        instance.save()
```

Comment:
I feel like all of this is too complicated, I don't really see why we should allow the API to create missing groups.
We need to talk about all this